src/commands/CamelRouteCommand.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict'
+
+import * as path from 'path';
+import * as fs from 'fs';
+import { TextDocument, window, workspace, WorkspaceFolder } from 'vscode';
+import { CamelJBangTask } from '../tasks/CamelJBangTask';
+
+import validFilename = require('valid-filename');
+
+export class CamelRouteCommand {
+
+	public static ID_COMMAND_CAMEL_ROUTE_JBANG = 'camel.jbang.routes.yaml';
+	public static YAML_FILE_EXTNSION = 'camel.yaml';
+
+	public async create(): Promise<void> {
+		const workspaceFolder = this.getWorkspaceFolder();
+		const providedName = await this.showInputBox(workspaceFolder);
+
+		if(providedName) {
+			const fileName = this.getFullName(providedName, CamelRouteCommand.YAML_FILE_EXTNSION);
+			const filePath = this.computeFullPath(workspaceFolder, fileName);
+
+			await new CamelJBangTask(workspaceFolder, fileName).execute();
+			await this.openFileInTextEditor(filePath);
```

Comment:
I think that we do not want to force opening in Text Editor and let it pick the preferred editor configured by the user. So that it can open by default with the Kaoto editor for instance. (sorry I should have thought about it on the first review).
```suggestion
			const uri = Uri.file(filePath);
			await commands.executeCommand('vscode.open', uri);
```